alf/optimizers/optimizers.py

```diff
@@ -501,6 +507,28 @@ def load_state_dict(self, state_dict):
                         state['rng_state'] = state['rng_state'].to(
                             self._rng_state_device).byte()
 
+    @common.add_method(NewCls)
+    def __setstate__(self, state):
+        def _move_key_as_the_first(d, key):
+            return {key: d.pop(key)}
+
+        for param_group in self.param_groups:
+            for p in param_group['params']:
+                p_state = self.state[p]
+                if 'step' not in p_state:
+                    # Some of the optimizers such as Adam/AdamW/ASGD etc in higher version
+                    # torch require 1) the presence of 'step' in state, and 2) it appears
+                    # as the first key of the state dictionary. Therefore we explicitly
+                    # create it if it does not exist and make it the key that comes first
+                    p_state.update({'step': 0})
+                    p_state = {
+                        **_move_key_as_the_first(p_state, 'step'),
+                        **p_state
+                    }
+                    self.state[p] = p_state
```

Comment:
can be simplified as: `self.state[p] = {'step': 0, **p_state}`

Reply:
Simplified. Thanks for the suggestion.